src/UseCases/AddComment/AddCommentUseCase.php

```diff
@@ -46,6 +51,10 @@ public function addComment( AddCommentRequest $addCommentRequest ): AddCommentRe
 
 		$donation->addComment( $this->newCommentFromRequest( $addCommentRequest ) );
 
+		if ( !$this->commentTextPassesValidation( $addCommentRequest->getCommentText() ) ) {
```

Comment:
I know. I did not see a way to do it only once and still have both checks happen, without adding mutable state. My thinking here is that the code is slightly simpler without the state, and that the performance difference is not relevant (no expensive resources are used). If you disagree with the later, or see a way to do it once without making the code more complicated, I'd love to hear it.